bindings/rust/src/lib.rs

```diff
@@ -365,6 +365,23 @@ impl ExecutionResult {
             None
         }
     }
+
+    pub fn typed_value(&self, expected_type: sys::FizzyValueType) -> Option<TypedValue> {
+        assert!(!self.0.trapped);
+        if self.0.has_value {
```

Comment:
Now the decision what to return is based on both `expected_type` and `self.0.has_value`

Maybe better change this to `if expected_type != sys::FizzyValueTypeVoid` and assert inside `assert!(self.0.has_value)`. Or even make it part of one bigger `match expected_type`.

Also maybe rename `expected_type` to just `type`, otherwise it looks like it's mainly not a conversion, but a check / test.

Reply:
Pushed a commit flipping conditions, but it doesn't make much of a difference. `assert!`'s are always in the binary, `debug_assert!`s aren't.

`type` is a keyword, hence `expected_type`.